marimo/_ast/cell.py

```diff
@@ -148,111 +144,211 @@ def set_status(self, status: CellStatusType) -> None:
         CellOp.broadcast_status(cell_id=self.cell_id, status=status)
 
 
-CellFuncType = Callable[..., Optional[Tuple[Any, ...]]]
-# Cumbersome, but used to ensure function types don't get erased in decorators
-# or creation of CellFunction
-CellFuncTypeBound = TypeVar(
-    "CellFuncTypeBound",
-    bound=Callable[..., Optional[Tuple[Any, ...]]],
-)
+@dataclasses.dataclass
+class Cell:
```

Comment:
The new `Cell` class, to be added to the public API. It also replaces `CellFunction` throughout our codebase

Reply:
should we add a doc for the `Cell` class?